src/python/dxpy/sugar/proc.py

```diff
@@ -0,0 +1,289 @@
+import shlex
+import subprocess
+
+from dxpy.sugar import STR_TYPES, info
+
+
+def run(
+    cmd, return_output=False, output_file=None, shell=True, block=True, echo=True,
+    **kwargs
+):
+    """
+    Execute a command using a subprocess.
+
+    Args:
+        cmd (str or list): Command to be executed using subprocess. Input of
+            type 'list' is recommended. When input is of type 'string',
+            command is executed using /bin/bash and shell=True is specified.
+        return_output (bool): If command output should be returned.
+        output_file (str): Name of file to which command stdout should be redirected.
+        shell: Whether to execute the command using the shell.
+        block: Whether to block until the process is complete. If True, `return_output`
+            is ignored.
+        echo: Whether to echo the command to the log before executing it.
+        kwargs: Additional arguments passed to check_output or check_call.
+
+    Note:
+        `return_output` and `output_file` paramters are mutually exclusive.
+        If output_file exists, it will be overwritten.
+
+    Returns:
+        * If `block=False`, returns the Popen for the running process.
+        * If return_output is provided, the output string is returned.
+        * If output_file is provided, stdout is redirected to the specified file
+        and the function returns None.
+
+    Raises:
+        CalledProcessError may be raised if the command failed to execute.
+
+    Examples:
+        run_cmd(['echo', 'hello world'], output_file='test1.txt')
+    """
+    executable = None
+    if shell:
+        executable = "/bin/bash"
```

Comment:
Often a command will leverage bash-specific syntax. I will make the 'shell' argument be able to take True, False, or a string specifying the executable. If True, it will default to bash.